src/coreclr/inc/clrconfignocache.h

```diff
@@ -0,0 +1,107 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+//
+// --------------------------------------------------------------------------------------------------
+// clrconfignocache.h
+//
+// Logic for resolving configuration names.
+//
+
+#ifndef _CLRCONFIGNOCACHE_H_
+#define _CLRCONFIGNOCACHE_H_
+
+#ifndef StrLen
+#define StrLen(STR) ((sizeof(STR) / sizeof(STR[0])) - 1)
+#endif // !StrLen
+
+// Config prefixes
+#define COMPLUS_PREFIX_A "COMPlus_"
+#define COMPLUS_PREFIX W("COMPlus_")
+#define LEN_OF_COMPLUS_PREFIX StrLen(COMPLUS_PREFIX_A)
+
+#define DOTNET_PREFIX_A "DOTNET_"
+#define DOTNET_PREFIX W("DOTNET_")
+#define LEN_OF_DOTNET_PREFIX StrLen(DOTNET_PREFIX_A)
+
+class CLRConfigNoCache
+{
+    const char* _value;
+
+    CLRConfigNoCache() = default;
+    CLRConfigNoCache(LPCSTR cfg) : _value { cfg }
+    { }
+
+public:
+    bool IsSet() const { return _value != NULL; }
+
+    LPCSTR AsString() const
+    {
+        _ASSERTE(IsSet());
+        return _value;
+    }
+
+    bool TryAsInteger(int radix, DWORD& result) const
+    {
+        _ASSERTE(IsSet());
+
+        errno = 0;
+        LPSTR endPtr;
+        result = strtoul(_value, &endPtr, radix);
+        bool fSuccess = (errno != ERANGE) && (endPtr != _value);
+        return fSuccess;
+    }
+
+    friend CLRConfigNoCache GetCLRConfigNoCache(LPCSTR cfg, bool noPrefix);
+};
+
+// This function is not a member of CLRConfigNoCache and marked inline so linkage
+// is relative to the compilation unit it is included in. The linkage concerns here
+// are due to usage in the PAL regarding getenv.
+inline CLRConfigNoCache GetCLRConfigNoCache(LPCSTR cfg, bool noPrefix = false)
+{
+    if (cfg == nullptr)
+        return {};
+
+    char nameBuffer[64];
+    const char* fallbackPrefix = NULL;
+    const size_t namelen = strlen(cfg);
+
+    if (noPrefix)
+    {
+        if (namelen >= _countof(nameBuffer))
+        {
+            _ASSERTE(!"Environment variable name too long.");
+            return {};
+        }
+
+        *nameBuffer = W('\0');
+    }
+    else
+    {
+        bool dotnetValid = namelen < (size_t)(_countof(nameBuffer) - 1 - LEN_OF_DOTNET_PREFIX);
+        bool complusValid = namelen < (size_t)(_countof(nameBuffer) - 1 - LEN_OF_COMPLUS_PREFIX);
+        if (!dotnetValid || !complusValid)
```

Comment:
I don't understand what is being stated/asked here. Could you elaborate?

Reply:
If DOTNET_ has enough buffer size, and COMPLUS_ doesn't, we error out. But actually sounds fine, you'd need a really long variable name to get there now that I see the 64.

Reply:
This is ensuring we can use either as a prefix with the name. I guess we could use the largest one but I'd prefer to keep this as is because it establishes a pattern and the majority of the computation here is compile time constant.